### ISSUE TITLE
cue export could unify packages split across directories

### ISSUE THREAD

**Issue by jack-bischoff** [cue-lang/cue#2018]
**Is your feature request related to a problem? Please describe.**
We store our configuration written in CUE in n number of child directories. Each child directory only has a depth of 1 but could have any number of cue files. Every cue file shares the package `mypkg`. In the parent directory we have a top level file called exports also in the `mypkg` package which holds an struct called exports. 

Like this:
```
---exports.cue---
exports: {}
```

Within each cue file we "export" the containing struct like this:

```
--- test1.cue---
myUniqueStruct1: {...}
exports: myUniqueStruct1
```

```
---test2.cue---
myUniqueStruct2: {...}
exports: myUniqueStruct2
```

Running cue export ./path/to/dir returns the expected result of:
```
{exports: {
  myUniqueStruct: {}
  myUniqueStruct2: {}
  ...
}}
```
However, running cue export ./path/to/dir1 ./path/to/dir2 returns a (personally) unexpected result of:
```
{
  exports: {
    dir1Struct1: {}
    dir1Struct2: {}
    ...
  }
}
{
  exports: {
    dir2Struct1: {}
    dir2Struct2: {}
    ...
}
```

The cue tool seems to create an "unification instance" for every path from the root to the leaves-containing node. 

**Describe the solution you'd like**
An opt-in flag, or some kind of documentation around a pattern, to unify shared packages that may existed laterally through a directory tree. The ultimate result would look like this:
```
{
  exports: {
    dir1Struct1: {}
    dir1Struct2: {}
    dir2Struct1: {}
    dir2Struct2: {}
    ...
  }
}
```

We originally considered shelling out to the cue tool to handle unification. Having everything in one data structure like that would have made parsing it slightly more convenient.

**Describe alternatives you've considered**
We also considered coalescing those objects in our CLI using `cue/load`. This was ultimately what we settled on since users need to interact with our tool anyways. 

Nevertheless, we think having cross-directory unification could prove helpful  as a feature either in the `cue/load` go package or `cue export`.

**Comment by jpluscplusm** [comment]
I agree: this is interesting behaviour, and I /suppose/ it's one reading of what's implied by `cue export --help`:

> \# multiple files: these are combined at the top-level. Order doesn't matter.

On the other hand, "combined at the top-level" could read as "objects co-existing at the top level are unified, just as if they were entries in different files but the same package being unified during 'normal' operation, with no cross-directory stuff going on".  In other words, I think the help text isn't actually telling us clearly what's going to happen, and either you've expected the reading that's /not/ happening; or there's a bug.

Either way, I've found a dirty hack that might help you with your immediate problem, @jack-bischoff. I've tested it with v0.4.3 and 0.5.0-beta.2:

```
$ find . -type f
./1/file.cue
./2/file.cue
$ tail -n 1000 */file.cue
==> 1/file.cue <==
package main

x: one: true

==> 2/file.cue <==
package main

x: two: false
$ cue export --out cue ./1/ ./2/ | cue export -
{
    "x": {
        "one": true,
        "two": false
    }
}
```

Hope this helps!

**Comment by myitcv** [comment]
@jack-bischoff thanks for raising and apologies for the delay in updating on this issue.

Your use case very much chimes with other use cases where we've seen a similar need. 

I am adding this comment in the middle of a large issue gardening session; we will revisit feature requests in a later pass.

**Comment by mvdan** [comment]
I think it makes sense to close this in favor of https://github.com/cue-lang/cue/issues/2486, which tracks work in this space. This issue is not resolved nor a duplicate per se, but it's best to focus attention on a single open issue on this topic. It's also tricky to find the right one when we have a few open at the same time which are fairly similar :) 

Having this issue works well as an archive to explain Vector's use case, though, so it's still useful when linked from 2486.
